### ISSUE TITLE
Unit tests for the Boundary clas

### ISSUE THREAD

**Issue by JMorado** [NOC-MSM/pyBDY#138]
As a starting point for having unit tests in PyBDY, it could be interesting to focus on the `Boundary` class defined in `nemo_bdy_gen_c.py`. From discussions with James, we concluded that this class is rather self-contained, which makes it ideal to start with.

However, I believe that before writing unit tests for the methods of this class, the class itself should be slightly refactored. The `__init__` constructor method is currently performing too many operations. Most of the code in this constructor should be moved to private methods. Additionally, I plan to document the methods, which could serve as an example for the future.

Are there any objections to refactoring this class? Does anyone have other ideas on how to proceed?

Many thanks.

**Comment by JMorado** [comment]
@rdPatmore Yes, exactly. Then we can also create unit tests for these private methods to ensure that initialization goes as expected.

**Comment by rdPatmore** [comment]
@JMorado Ok, I'll put myself down as an assignee, with a view to do some refactoring.

**Comment by rdPatmore** [comment]
@JMorado I have created a branch for this issue and completed some initial refactoring. We may wish to make further modifications to this new structure. The functions could also do with better documentation (i.e. more detailed docstrings).

**Comment by JMorado** [comment]
@rdPatmore, I have now finished the refactoring process, added appropriate docstrings, and included type hints where necessary.

I think we are now ready to start writing the unit tests for every method. This will also tell us if further modifications to the class are needed.

**Comment by JMorado** [comment]
@rdPatmore, we've got a few functional/unit tests up and running. If you're interested, we could catch up and chat about our next steps. Let me know!

**Comment by rdPatmore** [comment]
@JMorado great! I've had a quick look at the code. Much improved. Happy to chat at somepoint over the next week.

**Comment by JMorado** [comment]
@rdPatmore, I have completed this task. I will create a PR for this issue.

@jdha, regarding the docstrings, see [function_with_pep484_type_annotations example](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html) for an example close to what we are doing. In my opinion, replicating the types in the docstring might be unnecessary since they are already specified through the type annotations. Currently, we are not duplicating the return type in the docstring in order to adhere to the NumPy docstring style. However, we could consider adding this duplication at a later time.
